aten/src/ATen/native/cuda/TensorCompare.cu

```diff
@@ -43,7 +43,7 @@ void clamp_kernel_impl(TensorIteratorBase& iter) {
   AT_DISPATCH_ALL_TYPES_AND2(kHalf, kBFloat16, iter.common_dtype(), "clamp_cuda", [&] {
     gpu_kernel(iter, []GPU_LAMBDA(scalar_t v, scalar_t lower, scalar_t upper) -> scalar_t {
       // Propagate nan, which doesn't propagate automatically for ROCm
-      if (at::_isnan(v)) {
+      if (at::_isnan(v) || at::_isnan(lower) || at::_isnan(upper)) {
```

Comment:
The line below should be changed to a quiet nan?

Reply:
Oh you are right, except cuda doesn't like `std::numeric_limits<scalar_t>::quiet_NaN();` so it'll be a few ifs, will do.